File: services/device/src/main/java/com/example/iter/device/dto/response/RentalScheduleItemResponse.java
package com.example.iter.device.dto.response;

import com.example.iter.reservation.api.RentalScheduleItem;
import com.example.iter.reservation.api.RentalStatus;

import java.time.LocalDate;

public record RentalScheduleItemResponse(
        Long rentalId,
        LocalDate startDate,
        LocalDate endDate,
        RentalStatus status
) {
    public static RentalScheduleItemResponse from(RentalScheduleItem item) {
        return new RentalScheduleItemResponse(
                item.rentalId(),
                item.startDate(),
                item.endDate(),
                item.status()
        );
    }
}

File: services/device/src/main/kotlin/com/example/iter/device/dto/response/RentalScheduleItemResponse.kt
package com.example.iter.device.dto.response

import com.example.iter.reservation.api.RentalScheduleItem
import com.example.iter.reservation.api.RentalStatus

import java.time.LocalDate

data class RentalScheduleItemResponse(
    val rentalId: Long,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val status: RentalStatus,
) {
    companion object {
        @JvmStatic
        fun from(item: RentalScheduleItem): RentalScheduleItemResponse =
            RentalScheduleItemResponse(
                rentalId = item.rentalId(),
                startDate = item.startDate(),
                endDate = item.endDate(),
                status = item.status(),
            )
    }
}
